fetch.bs

```diff
@@ -2940,67 +2940,99 @@ run these steps:</p>
-"<code>no-cors</code>".
+response <a for=/>header</a> to control the contexts into which a given resource may be loaded. The
+restrictions asserted by this header apply to "<code>no-cors</code>" subresource requests and to
+<a>navigation requests</a> that target a frame or a window that's accessible from the request's
+initiator.
 
 <p>Its <a for=header>value</a> <a>ABNF</a>:
 
 <pre><code class=lang-abnf>
-Cross-Origin-Resource-Policy     = %s"same-origin" / %s"same-site" ; case-sensitive
+Cross-Origin-Resource-Policy     = %s"same-origin" / %s"same-site" / %s"cross-site" ; case-sensitive
 </code></pre>
 
 <p>To perform a <dfn>cross-origin resource policy check</dfn>, given a <var>request</var> and
 <var>response</var>, run these steps:</p>
 
 <ol>
- <li><p>If <var>request</var>'s <a for=request>mode</a> is not "<code>no-cors</code>", then return
- <b>allowed</b>.
-
  <li>
-  <p>If <var>request</var>'s <a for=request>origin</a> is <a>same origin</a> with
-  <var>request</var>'s <a for=request>current URL</a>'s <a for=url>origin</a>, then return
-  <b>allowed</b>.
+  <p>Return <b>allowed</b> if either of the following statements is true:
+  <ul class=brief>
+   <li><p><var>request</var>'s <a for=request>mode</a> is "<code>same-origin</code>",
+   "<code>cors</code>", or "<code>websocket</code>".
 
-  <p class="note no-backref">While redirects that carry  a
-  `<a http-header><code>Cross-Origin-Resource-Policy</code></a>` header are checked, redirects
-  without such a header resulting in <var>response</var> do not contribute to this algorithm. I.e.,
-  <var>request</var>'s <a for=request>tainted origin flag</a> is not checked.
+   <li><p><var>request</var> is a <a>navigation request</a> whose <a for=request>reserved client</a>
+   is an <a for=/>environment</a> whose <a for=environment>target browsing context</a> is neither a
+   <a>nested browsing context</a> nor an <a>auxiliary browsing context</a>.
+  </ul>
 
  <li>
   <p>Let <var>policy</var> be the result of <a for="header list">getting</a>
   `<a http-header><code>Cross-Origin-Resource-Policy</code></a>` from <var>response</var>'s
   <a for=response>header list</a>.
 
   <p class=note>This means that `<code>Cross-Origin-Resource-Policy: same-site, same-origin</code>`
-  ends up as <b>allowed</b> below as it will never match anything. Two or more
-  `<a http-header><code>Cross-Origin-Resource-Policy</code></a>` headers will have the same effect.
-
- <li><p>If <var>policy</var> is `<code>same-origin</code>`, then return <b>blocked</b>.
+  ends up as <b>allowed</b> as it will not match any of the explicit options in the switch below.
+  Two or more `<a http-header><code>Cross-Origin-Resource-Policy</code></a>` headers delivered in
+  the same response will similarly fall through to <b>allowed</b> below.
 
  <li>
-  <p>If the following are true
+  <p>If <var>policy</var> is null, and <a>request</a>'s <a for=request>client</a>'s
```

Comment:
Is this what we want? I kinda feel like an erroneous policy should fail closed when this TBD boolean is set.

Reply:
I want to be able to support new values for this header in the future (for example, a list of origins a la https://github.com/whatwg/fetch/issues/760). I'm happy to tighten this up, but I'd like to figure out how to do so in a forward-compatible way.

Reply:
I guess the question then is what the migration strategy has to be. In equivalent cases we've failed for potentially new syntax and new servers will have to update (e.g., all the recentish and proposed CORS changes).